hooks/tk-flame_actions.py

```diff
@@ -0,0 +1,383 @@
+        else:
+            raise FlameActionError("File not found on disk - '%s'" % batch_path)
+
+    def _import_clip(self, sg_publish_data):
+        """
+        Imports a clip to the current Batch Group.
+
+        :param sg_publish_data: Shotgun data dictionary with all the standard publish fields.
+        :type sg_publish_data: dict
+        """
+
+        clip_path = self.get_publish_path(sg_publish_data)
+
+        # Only load the clip if it exists
+        if clip_path and os.path.exists(clip_path):
+            flame.batch.go_to()
+            flame.batch.import_clip(clip_path, SCHEMATIC_REEL)
+            flame.batch.organize()
+        elif '%' in clip_path:
+            try:
+                # Special case parsing for frames, attempts to expand
+                temp_filters = [["id", "is", sg_publish_data["version"]["id"]]]
+                temp_fields = ["frame_range"]
+                temp_type = "Version"
+
+                temp_info = self.parent.shotgun.find_one(
+                    temp_type, filters=temp_filters, fields=temp_fields
+                )
+
+                # Unfortunately, we can't do simple formatting here as %<num>d
+                # old style Python formatting does not support getting a frame
+                # range. Thus we need to parse it ourselves
+                new_path = self._handle_frame_range(
+                    clip_path, temp_info["frame_range"]
+                )
+
+                if new_path and len(new_path) != 0:
+                    clip_path = new_path
+
+                flame.batch.go_to()
+                flame.batch.import_clip(clip_path, SCHEMATIC_REEL)
+                flame.batch.organize()
+            except Exception:
+                raise FlameActionError("File not found on disk - '%s'" % clip_path)
+        else:
+            raise FlameActionError("File not found on disk - '%s'" % clip_path)
+
+    def _import_batch_group(self, sg_publish_data):
+        """
+        Imports a Batch Group from Shotgun into Flame.
+
+        :param sg_publish_data: Shotgun data dictionary with all the standard publish fields.
+        :type sg_publish_data: dict
+        """
+        # Determines published files
+        sg_filters = [["id", "is", sg_publish_data["id"]]]
+        sg_fields = ["sg_published_files",
+                     "code",
+                     "sg_cut_in",
+                     "sg_cut_out",
+                     "sg_versions",
+                     "sg_sequence"
+                     ]
+
+        sg_type = sg_publish_data["type"]
+
+        sg_info = self.parent.shotgun.find_one(
+            sg_type, filters=sg_filters, fields=sg_fields
+        )
+
+        # Checks that we have the necessary info to proceed.
+        if not all(f in sg_info for f in sg_fields):
+            raise FlameActionError("Cannot load a Batch Group from Shotgun using this {}".format(sg_type))
+
+        published_files = self._get_paths_from_published_files(
```

Comment:
[minor] The naming here is a bit confusing. published_files is published_file_paths. Also _get_batch_path_from_published_files could call _get_paths_from_published_files directly.